File: thorium/llmGenerated/testAll.py
import pandas as pd
csv = "data.csv"
table = pd.read_csv(csv)
print(table.to_string())
values = "pierre, 21, rennes,GMA".split(",")
new_row = pd.DataFrame([values],columns=table.columns)
table = pd.concat([table,new_row], ignore_index=True)
new_values = ["serge,21,rennes,GMA","paul, 22, paris,GMA","herve, 23, lyon,INFO"]
for row in new_values:
    values = row.split(",")
    new_row = pd.DataFrame([values],columns=table.columns)
    table = pd.concat([table,new_row], ignore_index=True)
print(table.to_string())
table['age_SUM'] = table["age"].sum()
table['age_COUNT'] = table.shape[0]
table.loc[1] = ["Hugo",22,"Lyon","INFO"]
table.iloc[1, 3] = "GMA"
table.at[1, 'age'] = 40
print(table.to_string())
table = table[(table['department'] == "INFO") & (table['age'] >= 20)]
print(table.to_string())
table = table.drop(1)
table = table.drop("department", axis=1)
table = table.drop([1,2])
print(table.to_string())
table = table["name"]
table = table[["name","age"]]
table.to_csv("data2.csv",index=False)

File: thorium/llmGenerated/testAll.R
csv <- "data.csv"
table <- read.csv(csv, stringsAsFactors = FALSE)
print(table)
table[nrow(table) + 1,] <- c("pierre",21," rennes","GMA")
rows <- list(c("serge",21,"rennes","GMA"),c("paul",22," paris","GMA"),c("herve",23," lyon","INFO"))
for (row in rows){
    table[nrow(table) + 1,] <- row
}
print(table)
table <- cbind(table, age_SUM = rep(sum(table$age),length.out=length(table$age)))
table <- cbind(table, age_COUNT = rep(length(table$age),length.out=length(table$age)))
table[2,] <- c("Hugo",22,"Lyon","INFO")
table[2, 4] <- "GMA"
table[2, "age"] <- 40
print(table)
table <- subset(table, department == "INFO" & "age" >= 20)
print(table)
table <- table[-2,]
table <- table[,!names(table) %in% c("department")]
table <- table[-c(2,3),]
print(table)
table <- data.frame(list(name = table[,c("name")]))
table <- table[,c("name","age")]
write.csv(table, "data2.csv", row.names=FALSE, quote=FALSE)